### PR TITLE
[dynamo][benchmarks] add option to force amp to use bfloat16 instead of float16

### PR DESCRIPTION
**Pull request by xmfan**
Stack from [ghstack](https://github.com/ezyang/ghstack/tree/0.12.0) (oldest at bottom):
* __->__ #169449

For models which hardcode bf16 like modded-nanogpt

Tested on `python benchmarks/dynamo/torchbench.py --performance --training --amp --backend inductor --device cuda --only modded_nanogpt --disable-cudagraphs` w/ https://github.com/pytorch/benchmark/pull/2660

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @kadeng @chauhang @amjames @Lucaskabela @jataylo

### CODE REVIEW

**Review by Lucaskabela — APPROVED** [review]
LGTM - will this move the benchmark's expected value/accuracy at all?
